WireAPI/Sources/WireAPI/APIs/ConversationsAPI/ConversationsAPIV3.swift

```diff
@@ -20,4 +20,143 @@ import Foundation
+            notFound: notFound,
+            failed: failed
+        )
+    }
+}
+
+// MARK: -
+
+private struct ConversationV3: Decodable, ToAPIModelConvertible {
+    enum CodingKeys: String, CodingKey {
+        case access
+        case accessRole = "access_role"
+        case accessRoleV2 = "access_role_v2"
+        case creator
+        case epoch
+        case id
+        case lastEvent = "last_event"
+        case lastEventTime = "last_event_time"
+        case members
+        case messageProtocol = "protocol"
+        case messageTimer = "message_timer"
+        case mlsGroupID = "group_id"
+        case name
+        case qualifiedID = "qualified_id"
+        case readReceiptMode = "receipt_mode"
+        case teamID = "team"
+        case type
+    }
+
+    var access: [String]?
+    var accessRoles: [String]?
+    var creator: UUID?
+    var epoch: UInt?
+    var id: UUID?
+    var lastEvent: String?
+    var lastEventTime: String?
+    var legacyAccessRole: String?
+    var members: QualifiedConversationMembers?
+    var messageProtocol: String?
+    var messageTimer: TimeInterval?
+    var mlsGroupID: String?
+    var name: String?
+    var qualifiedID: QualifiedID?
+    var readReceiptMode: Int?
+    var teamID: UUID?
+    var type: Int?
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        access = try container.decodeIfPresent([String].self, forKey: .access)
+        creator = try container.decodeIfPresent(UUID.self, forKey: .creator)
+        epoch = try container.decodeIfPresent(UInt.self, forKey: .epoch)
+        id = try container.decodeIfPresent(UUID.self, forKey: .id)
+        lastEvent = try container.decodeIfPresent(String.self, forKey: .lastEvent)
+        lastEventTime = try container.decodeIfPresent(String.self, forKey: .lastEventTime)
+        members = try container.decodeIfPresent(QualifiedConversationMembers.self, forKey: .members)
+        messageProtocol = try container.decodeIfPresent(String.self, forKey: .messageProtocol)
+        messageTimer = try container.decodeIfPresent(TimeInterval.self, forKey: .messageTimer)
+        mlsGroupID = try container.decodeIfPresent(String.self, forKey: .mlsGroupID)
+        name = try container.decodeIfPresent(String.self, forKey: .name)
+        qualifiedID = try container.decodeIfPresent(QualifiedID.self, forKey: .qualifiedID)
+        readReceiptMode = try container.decodeIfPresent(Int.self, forKey: .readReceiptMode)
+        teamID = try container.decodeIfPresent(UUID.self, forKey: .teamID)
+        type = try container.decodeIfPresent(Int.self, forKey: .type)
+
+        // v3 replaces the field "access_role_v2" with "access_role".
+        // However, since the format of update events does not depend on versioning,
+        // we may receive conversations from the `conversation.create` update event
+        // which still have both "access_role_v2" and "access_role" fields
```

Comment:
Perhaps we shouldn't use this struct for decoding update events containing conversations. If we didn't, would we still need this custom decoding?

Reply:
right, good catch 03fd6767da32b530f841978ffcafbdfa58dcaf9b

Reply:
I left a question at the same code in the V5 version of this, but I'll repeat it here. Do I understand correctly that this logic allows us to reuse the payload in both /conversation responses as well as in update event payloads? If so, then I would say they should have separate payload models, and here we just have the payload only for versioned api responses, meaning you could rely on the generated decodable conformance.

Reply:
Alright, fixed (see comment on V5): 57bf86d46884e9109257b3060ded1b0d9efc9900